app/src/main/java/org/zalando/nakadi/config/AsyncConfig.java

```diff
@@ -0,0 +1,18 @@
+package org.zalando.nakadi.config;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.task.AsyncTaskExecutor;
+import org.springframework.scheduling.concurrent.ConcurrentTaskExecutor;
+
+import java.util.concurrent.Executors;
+
+@Configuration
+public class AsyncConfig {
+    @Bean
+    @Qualifier("mvcAsyncTaskExecutor")
+    public AsyncTaskExecutor taskExecutor() {
+        return new ConcurrentTaskExecutor(Executors.newCachedThreadPool());
+    }
+}
```

Comment:
@a1exsh 
Is there a reason why the configuration got to different class?

Reply:
Not really.  I just want to make sure that we create the thread pool only once.  If there is a way to achieve that while keeping it in the `WebConfig` class, I'd take that happily.